contracts/YearnHarvest.sol

```diff
@@ -114,80 +105,141 @@ contract YearnHarvest {
-    are expected to call checkHarvestStatus() as a static, "view" call off-chain.
-    @dev The method relies on each strategy to determine whether it's ready for harvest
-    (via harvestTrigger()), and attempts a non-state-changing harvest() when the strategy
-    trigger returns TRUE.
-    Even if the trigger is TRUE and the simulated call to harvest() is succesful, certain
-    strategies may have not needed to be harvested (e.g., Total Assets == 0,
-    Last run < 10 minutes ago, etc.). This is why we encourage strategists that would like
-    to rely on yHarvest for automation to ensure their triggers are accurate to avoid
-    unnecesary calls to harvest and the cost associated with it.
-    @return canExec boolean indicating whether the strategy is ready to harvest()
-    @return execPayload call data used by Gelato executors to call harvestStrategy(). It
+    @notice Create Gelato job that will monitor new strategies in Yearn Lens. When
+    a new startegy is detected with its keeper set as yHarvest, yHarvest creates a
+    gelato job for it. 
+    */
+    function initiateStrategyMonitor() external onlyAuthorized {
+        jobIds[address(this)] = ops.createTaskNoPrepayment(
+            address(this), // `execAddress`
+            this.createJobAndPay.selector, // `execSelector`
+            address(this), // `resolverAddress`
+            abi.encodeWithSelector(this.checkNewStrategies.selector),
+            feeToken
+        );
+    }
+
+    /**
+    @notice Creates Gelato job for a strategy. The job and resolver address are the same -- the
+    Yearn Harvest contract. Updates `jobIds`, which we use to log events and
+    to manage the job.
+    @param strategyAddress Strategy Address for which a job will be created
+    */
+    function createJobAndPay(address strategyAddress) external onlyKeepers {
+        createJob(strategyAddress);
+
+        // `gelatoFee` and `gelatoFeeToken` are state variables in the gelato contract that
+        // are temporarily modified by the executors before executing the payload. They are
+        // reverted to default values when the gelato contract exec() method wraps up.
+        (uint256 gelatoFee, address gelatoFeeToken) = ops.getFeeDetails();
+
+        require(gelatoFeeToken == feeToken, "!token"); // dev: gelato not using intended token
+        require(gelatoFee <= maxFee, "!fee"); // dev: gelato executor overcharnging for the tx
+
+        // Pay Gelato for the service.
+        payKeeper(gelatoFee);
+    }
+
+    /**
+    @notice Like `createJobAndPay()` but without payment. We can use this function to 
+    manually create new jobs, if needed (for example, if we have to maually cancel any
+    and re-start it afterwards). 
+    @param strategyAddress Strategy Address for which a job will be created
+    */
+    function createJob(address strategyAddress) public onlyKeepers {
+        jobIds[strategyAddress] = ops.createTaskNoPrepayment(
```

Comment:
I dropped that assert because at this point the payload must only include information from strategies that have the contract assigned as keeper. The risk of not asserting again here is that the Gelato contract is somehow used maliciously to create jobs. Do you see any other risk? I'm happy to add it back